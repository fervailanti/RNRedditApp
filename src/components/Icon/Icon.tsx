import React from 'react'
import { StyleProp, TextStyle } from 'react-native'
import { getIconFamily, IconFamily } from './iconFamilies'

interface IconProps {
  family?: IconFamily
  color?: string
  size?: number
  style?: StyleProp<TextStyle>
  name: string
}

const Icon: React.FC<IconProps> = ({
  family = 'MaterialCommunityIcons',
  color = '#999',
  size = 16,
  style,
  name,
}) => {
  const Component = getIconFamily(family)
  return <Component color={color} name={name} size={size} style={style} />
}

export default Icon

export type { IconFamily }
