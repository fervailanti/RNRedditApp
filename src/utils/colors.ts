export const colors = {
  primary: '#FF4400',
  black: '#000000',
  white: '#FFFFFF',
  gray: {
    dark: '#666666',
    light: '#999999',
  },
}
