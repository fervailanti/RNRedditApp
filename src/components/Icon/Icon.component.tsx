import { colors } from '@utils/colors'
import React from 'react'
import { StyleProp, TextStyle } from 'react-native'
import { getIconFamily, IconFamily } from './iconFamilies'

interface IconProps {
  family?: IconFamily
  color?: string
  size?: number
  style?: StyleProp<TextStyle>
  name: string
}

const Icon: React.FC<IconProps> = ({
  family = 'MaterialCommunityIcons',
  color = colors.gray.light,
  size = 16,
  style,
  name,
}) => {
  const Component = getIconFamily(family)
  return <Component color={color} name={name} size={size} style={style} />
}

export default Icon

export type { IconFamily }
