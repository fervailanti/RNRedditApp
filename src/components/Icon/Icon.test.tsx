import { render } from '@testing-library/react-native'
import { colors } from '@utils/colors'
import React from 'react'
import 'react-native'
import Icon from './Icon.component'

const props = {
  color: colors.primary,
  size: 16,
}

describe('Components > Icon', () => {
  it.each([
    // Case #1
    ['Entypo' as const, 'aircraft'],
    // Case #2
    ['Fontisto' as const, 'paw'],
    // Case #3
    ['AntDesign' as const, 'phone'],
    // Case #4
    ['EvilIcons' as const, 'plus'],
    // Case #5
    ['Feather' as const, 'activity'],
    // Case #6
    ['FontAwesome' as const, 'table'],
    // Case #7
    ['Foundation' as const, 'puzzle'],
    // Case #8
    ['Ionicons' as const, 'basket'],
    // Case #9
    ['MaterialCommunityIcons' as const, 'account'],
    // Case #10
    ['MaterialIcons' as const, 'drafts'],
    // Case #11
    ['Octicons' as const, 'plug'],
    // Case #12
    ['SimpleLineIcons' as const, 'like'],
    // Case #13
    ['Zocial' as const, 'android'],
  ])(
    '<Icon {...props} /> should render correctly every type of family',
    (family, name) => {
      const allProps = { ...props, family, name }
      const { toJSON } = render(<Icon {...allProps} />)
      expect(toJSON()).toMatchSnapshot()
    },
  )
})
