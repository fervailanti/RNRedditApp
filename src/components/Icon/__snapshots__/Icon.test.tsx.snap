// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > Icon <Icon {...props} /> should render correctly every type of family 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "#FF4400",
        "fontSize": 16,
      },
      undefined,
      Object {
        "fontFamily": "Entypo",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`Components > Icon <Icon {...props} /> should render correctly every type of family 2`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "#FF4400",
        "fontSize": 16,
      },
      undefined,
      Object {
        "fontFamily": "Fontisto",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`Components > Icon <Icon {...props} /> should render correctly every type of family 3`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "#FF4400",
        "fontSize": 16,
      },
      undefined,
      Object {
        "fontFamily": "anticon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`Components > Icon <Icon {...props} /> should render correctly every type of family 4`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "#FF4400",
        "fontSize": 16,
      },
      undefined,
      Object {
        "fontFamily": "EvilIcons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`Components > Icon <Icon {...props} /> should render correctly every type of family 5`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "#FF4400",
        "fontSize": 16,
      },
      undefined,
      Object {
        "fontFamily": "Feather",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`Components > Icon <Icon {...props} /> should render correctly every type of family 6`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "#FF4400",
        "fontSize": 16,
      },
      undefined,
      Object {
        "fontFamily": "FontAwesome",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`Components > Icon <Icon {...props} /> should render correctly every type of family 7`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "#FF4400",
        "fontSize": 16,
      },
      undefined,
      Object {
        "fontFamily": "fontcustom",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`Components > Icon <Icon {...props} /> should render correctly every type of family 8`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "#FF4400",
        "fontSize": 16,
      },
      undefined,
      Object {
        "fontFamily": "Ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`Components > Icon <Icon {...props} /> should render correctly every type of family 9`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "#FF4400",
        "fontSize": 16,
      },
      undefined,
      Object {
        "fontFamily": "Material Design Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  󰀄
</Text>
`;

exports[`Components > Icon <Icon {...props} /> should render correctly every type of family 10`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "#FF4400",
        "fontSize": 16,
      },
      undefined,
      Object {
        "fontFamily": "Material Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`Components > Icon <Icon {...props} /> should render correctly every type of family 11`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "#FF4400",
        "fontSize": 16,
      },
      undefined,
      Object {
        "fontFamily": "Octicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`Components > Icon <Icon {...props} /> should render correctly every type of family 12`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "#FF4400",
        "fontSize": 16,
      },
      undefined,
      Object {
        "fontFamily": "simple-line-icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`Components > Icon <Icon {...props} /> should render correctly every type of family 13`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "#FF4400",
        "fontSize": 16,
      },
      undefined,
      Object {
        "fontFamily": "zocial",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;
